src/main/scala/com/microsoft/hyperspace/util/ResolverUtils.scala

```diff
@@ -71,4 +78,87 @@ object ResolverUtils {
       availableStrings: Seq[String]): Option[Seq[String]] = {
     Some(requiredStrings.map(resolve(spark, _, availableStrings).getOrElse { return None }))
   }
+
+  /**
+   * Finds all resolved strings for requiredStrings, from the given logical plan. Returns
+   * optional seq of resolved strings if all required strings are resolved, otherwise None.
+   *
+   * @param spark Spark session.
+   * @param requiredStrings List of strings to resolve.
+   * @param plan Logical plan to resolve against.
+   * @return Optional ListMap of tuples of resolved name string and nested state boolean
+   *         if all required strings are resolved. Else, None.
+   */
+  def resolve(
+      spark: SparkSession,
+      requiredStrings: Seq[String],
+      plan: LogicalPlan): Option[ListMap[String, Boolean]] = {
+    val schema = plan.schema
+    val resolver = spark.sessionState.conf.resolver
+    val resolved = requiredStrings.map { requiredField =>
+      plan
+        .resolveQuoted(requiredField, resolver)
+        .map { expr =>
+          val resolvedColNameParts = extractColumnName(expr)
+          validateResolvedColumnName(requiredField, resolvedColNameParts)
+          getColumnNameFromSchema(schema, resolvedColNameParts, resolver)
+            .foldLeft(("", false)) { (acc, i) =>
+              val name = Seq(acc._1, i._1).filter(_.nonEmpty).mkString(".")
+              val isNested = acc._2 || i._2
+              (name, isNested)
+            }
+        }
+        .getOrElse { return None }
+    }
+    Some(ListMap(resolved: _*))
+  }
+
+  // Extracts the parts of a nested field access path from an expression.
+  private def extractColumnName(expr: Expression): Seq[String] = {
+    expr match {
+      case a: Attribute =>
+        Seq(a.name)
+      case g @ GetStructField(_, _, Some(name)) =>
+        extractColumnName(g.child) :+ name
+      case _: GetArrayStructFields =>
+        // TODO: Nested arrays will be supported later
+        throw HyperspaceException("Array types are not supported.")
+      case _: GetMapValue =>
+        // TODO: Nested maps will be supported later
+        throw HyperspaceException("Map types are not supported.")
+      case Alias(nested: ExtractValue, _) =>
+        extractColumnName(nested)
+    }
+  }
+
+  // Validate the resolved column name by checking if nested columns have dots in its field names.
+  private def validateResolvedColumnName(
+      origCol: String,
+      resolvedColNameParts: Seq[String]): Unit = {
+    if (resolvedColNameParts.length > 1 && resolvedColNameParts.exists(_.contains("."))) {
```

Comment:
Could you add this condition to here? from @imback82's comment
https://github.com/microsoft/hyperspace/pull/379/files/4f8e3928170fcee0c1813ead6d9ab1521548c278#r599699752 
```scala
if (resolvedColNameParts.length == 1 && resolvedColNameParts.head.contains(".")) {
  throw HyperspaceException(
    s"Hyperspace does not support the top-level column whose name contains dots: $origCol")
}
```

Reply:
Not needed anymore. Now, we support top level fields with `.` (dots).

Reply:
> Now, we support top level fields with `.` (dots). So it is good as it is.

Does this work?

```scala
    import spark.implicits._
    Seq(1).toDF("a.b").write.mode("overwrite").parquet("/tmp")
    val df = spark.read.parquet("/tmp")
    hyperspace.createIndex(df, IndexConfig("i", Seq("`a.b`"), Nil))
```

Reply:
I should be able to handle this in my follow up

Reply:
#393 handles the above case correctly. (we need a test coverage though)